Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift

```diff
@@ -333,11 +333,17 @@ class EnhancedTrackingProtectionMenuVC: UIViewController {
     }
 
     private func updateViewDetails() {
+        heroImage.image = UIImage(named: "defaultFavicon")!
+        heroImage.tintColor = UIColor.theme.etpMenu.defaultImageTints
         if let favIconURL = viewModel.favIcon {
-            heroImage.sd_setImage(with: favIconURL, placeholderImage: UIImage(named: "defaultFavicon"), options: [], completed: nil)
-        } else {
-            heroImage.image = UIImage(named: "defaultFavicon")!
-            heroImage.tintColor = UIColor.theme.etpMenu.defaultImageTints
+            ImageLoadingHandler.shared.getImageFromCacheOrDownload(with: favIconURL,
+                                                                   limit: ImageLoadingConstants.NoLimitImageSize) { image, error in
+                guard error == nil, let image = image else {
+                    return
+                }
+
+                self.heroImage.image = image
```

Comment:
For `self` in `getImageFromCacheOrDownload` calls, do we need to weakify `self` here? (I won't make the comment in each function haha, you get the point).

Reply:
Ah good catch, it should be weak